platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -101,74 +112,45 @@ public Hash(final ByteBuffer byteBuffer, final DigestType digestType) {
      * @param other
      * 		the hash to copy
      */
-    public Hash(final Hash other) {
+    public Hash(@NonNull final Hash other) {
         if (other == null) {
             throw new IllegalArgumentException("other");
         }
 
         this.digestType = other.digestType;
         this.value = Arrays.copyOf(other.value, other.value.length);
-    }
-
-    protected Hash(
-            final byte[] value,
-            final DigestType digestType,
-            final boolean bypassSafetyCheck,
-            final boolean shouldCopy) {
-        if (value == null) {
-            throw new IllegalArgumentException("value");
-        }
-
-        if (digestType == null) {
-            throw new IllegalArgumentException("digestType");
-        }
-
-        if (value.length != digestType.digestLength()) {
-            throw new IllegalArgumentException("value: " + value.length);
-        }
-
-        this.digestType = digestType;
-
-        final byte[] valuePtr = (shouldCopy) ? Arrays.copyOf(value, value.length) : value;
-
-        if (bypassSafetyCheck) {
-            this.value = valuePtr;
-        } else {
-            // Check for all zeros & stop when first non-zero byte has been encountered
-            for (byte b : value) {
-                if (b != 0) {
-                    this.value = value;
-                    return;
-                }
-            }
-
-            // We throw an exception here because the value array contained all zero bytes
-            throw new EmptyHashValueException("Hash creation failed, hash is array of zeroes");
-        }
+        this.bytes = other.bytes;
     }
 
     /**
      * Get the byte array representing the value of the hash.
      *
      * @return the hash value
      */
-    public byte[] getValue() {
+    public @NonNull byte[] getValue() {
         return value;
     }
 
+    /**
+     * @return a copy of the hash value as a byte array
+     */
+    public @NonNull byte[] copyToByteArray() {
+        return bytes.toByteArray();
+    }
+
```

Comment:
Why do both of these methods exist?

Reply:
`getValue` is supposed to be removed. I should have marked it as deprecated, I would rather not do that now since I need approvals from 8 teams for this PR

Reply:
`getValue` is supposed to be removed because we want to prevent access to the byte array. the new method explicitly says that a copy of the array is made, so that it is clear to anyone calling it that it has a performance impact